File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> map = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            int value = nums[i];
            int newTarget = target - value;
            
            if (map.containsKey(newTarget)) {
                return new int[] {map.get(newTarget), i};
            } else {
                map.put(value, i);
            }
        }
        
        return new int[] {-1, -1};
    }
}

File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
// nums = [2, 7, 11, 15], target = 9
var twoSum = function(nums, target) {
    let map = new Map();
    for (let i = 0; i < nums.length; i++) {
        let gap = target - nums[i];
        
        if (map.has(gap)) {
            return [map.get(gap), i];
        }
        
        map.set(nums[i], i);
    }
    
    return [-1, -1];
};